zeekpkg/manager.py

```diff
@@ -2969,6 +2970,54 @@ def install(self, pkg_path, version=""):
             LOG.warning('installing "%s": source package git repo is invalid', pkg_path)
             return f'failed to clone package "{pkg_path}": {error}'
 
+    def _validate_alias_conflict(self, pkg, metadata_dict):
+        """Check if there's an alias conflict.
+
+        If any of the installed packages aliases collide with the package's
+        name or its aliases, return a string describing the issue.
+
+        Args:
+            package (:class:`.package.Package`): the package to be installed
+
+            metadata_dict (dict): The metadata for the given package.
+                package.metadata may note be valid yet.
+
+        Returns:
+            str: empty string on success, else descriptive error message.
+        """
+        package_names = {}
+        alias_names = {}
+        for ipkg in self.installed_packages():
+            if ipkg.package == pkg:
+                continue
+
+            qn = ipkg.package.qualified_name()
+
+            package_names[ipkg.package.name] = qn
+            for ipkg_alias in ipkg.package.aliases():
+                alias_names[ipkg_alias] = qn
+
+        # Does the new package's name is the same as an existing alias?
+        if pkg.name in alias_names:
+            qn = alias_names[pkg.name]
```

Comment:
Nit: we could reduce the number of lookups here (this would be even more compact with python-3.8 assignment expressions, but we need >=python-3.7).

```suggestion
        qn = alias_names.get(pkg.name)
        if qn:
```

Same for the lookups in the loop below.

Reply:
If we had `:=` available, yes. Otherwise I find the `.get()` not as pretty.